pkg/kinds/lists.go

```diff
@@ -0,0 +1,70 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package kinds
+
+import (
+	"strings"
+
+	"github.com/pkg/errors"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// ListSuffix is the suffix expected for all Kubernetes collection resources.
+const ListSuffix = "List"
+
+// IsListGVK returns true if the kind has a "List" suffix.
+func IsListGVK(gvk schema.GroupVersionKind) bool {
+	return strings.HasSuffix(gvk.Kind, ListSuffix)
+}
+
+// ListGVKForItemGVK returns the item GroupVersionKind with "List" appended to
+// the kind.
+func ListGVKForItemGVK(gvk schema.GroupVersionKind) schema.GroupVersionKind {
+	gvk.Kind += ListSuffix
```

Comment:
Maybe this should check whether the GVK already has a List suffix to detect misuse, WDYT?

Reply:
I deliberately didn't validate here to make it easier to use without the error return. I think it's up to the caller whether they want to validate or not. Same for removing the suffix. But in general, yes, most places should probably check.